docs/DevGuide.md

```diff
@@ -72,13 +69,19 @@ Maestro.Web uses Azure AppConfiguration (AAC) to dynamically enable/disable auto
 - https://zimmergren.net/introduction-azure-app-configuration-store-csharp-dotnetcore/
 - https://docs.microsoft.com/en-us/azure/azure-app-configuration/howto-integrate-azure-managed-service-identity
 
+## Deploying a branch to Staging
+
+You can deploy your branch to the staging environment where E2E tests can be run using the following steps:
+- Notify others in the team that you are deploying to staging. The Staging environment is shared so please check if another `main` build is not running or others are not deploying to staging or about to merge a PR.
+- Push your branch to the Azure DevOps [dotnet-arcade-services](https://dev.azure.com/dnceng/internal/_git/dotnet-arcade-services) repository.
+- Run the [arcade-services-internal-ci](https://dev.azure.com/dnceng/internal/_build?definitionId=252&_a=summary) pipeline from your branch. You can unselect the secret rotation, approval and SDL stages if you want. Those are not required.
+- If a conflicting build starts to run, you can cancel one of them and restart them later so that only one pipeline runs the `Deploy` stage at once. The E2E tests will also fail if the deployed version does not match the one from the build.
```

Comment:
> The E2E tests will also fail if the deployed version does not match the one from the build.

We removed the test for the non-matching versions - https://github.com/dotnet/arcade-services/commit/eda3437d7ec61723ed40ff99ab838329a3a62d39
Or is the version match tested in some other way?

Reply:
Ah, thanks

```suggestion
- If a conflicting build starts to run, you can cancel one of them and restart them later so that only one pipeline runs the `Deploy` stage at once.
```